docs/quickstart-tenancy-and-apis.md

```diff
@@ -0,0 +1,314 @@
+
+```shell
+$ kubectl get clusterworkspacetypes
+NAME           AGE
+home           74m
+homebucket     74m
+homeroot       74m
+organization   74m
+root           74m
+team           74m
+universal      74m
+$ kubectl describe clusterworkspacetype/team
+Name:         team
+...
+API Version:  tenancy.kcp.dev/v1alpha1
+Kind:         ClusterWorkspaceType
+...
+Spec:
+  Default Child Workspace Type:
+    Name:  universal
+    Path:  root
+  Extend:
+    With:
+      Name:  universal
+      Path:  root
+  Limit Allowed Parents:
+    Types:
+      Name:  organization
+      Path:  root
+      Name:  team
+      Path:  root
+Status:
+  Conditions:
+    Status:                True
+    Type:                  Ready
+...
+```
+
+## Workspaces FAQs
+
+Q: Why do we have both `ClusterWorkspaces` and `Workspaces`?
+
+A: `Workspaces` are intended to be the user-facing resource, whereas `ClusterWorkspaces` is a low-level resource for kcp
+platform admins.
+
+`Workspaces` are actually a "projected resource", there is no such resource stored in etcd, instead it is served as a
+transformation of the `ClusterWorkspace` resource. `ClusterWorkspaces` contain details like shard assignment, which are
+low-level fields that users should not see.
+
+We are working on a change of this system behind the scenes. That will probably promote Workspaces to a normal,
+non-projected resource, and ClusterWorkspaces will change in its role.
+
+## Publish some APIs as a service provider
+
+kcp offers `APIExport` and `APIBinding` resources which allow a service provider operating in one workspace to offer its
+capabilities to service consumers in other workspaces.
+
+First we'll create an organization workspace, and then within that create a service provider workspace.
+
+```shell
+$ kubectl ws create wildwest --enter
+Workspace "wildwest" (type root:organization) created. Waiting for it to be ready...
+Workspace "wildwest" (type root:organization) is ready to use.
+Current workspace is "root:wildwest".
+$ kubectl ws create cowboys-service --enter
+Workspace "cowboys-service" (type root:universal) created. Waiting for it to be ready...
+Workspace "cowboys-service" (type root:universal) is ready to use.
+Current workspace is "root:wildwest:cowboys-service".
+```
+
+Then we'll use a CRD to generate an `APIResourceSchema` and `APIExport` and apply these within the service provider
+workspace.
+
+```shell
+$ ./bin/apigen --input-dir test/e2e/customresourcedefinition/ --output-dir .
+$ ls -1 *wildwest.dev.yaml
+apiexport-wildwest.dev.yaml
+apiresourceschema-cowboys.wildwest.dev.yaml
+
+$ kubectl apply -f apiresourceschema-cowboys.wildwest.dev.yaml
+apiresourceschema.apis.kcp.dev/v220920-6039d110.cowboys.wildwest.dev created
+$ kubectl apply -f apiexport-wildwest.dev.yaml
+apiexport.apis.kcp.dev/wildwest.dev created
+```
+
+You can think of an `APIResourceSchema` as being equivalent to a CRD, and an `APIExport` makes a set of schemas
```

Comment:
APIRS === a snapshot of a CRD at one point in time. Makes the choice to move from one schema to another an explicit choice

Reply:
Can adjust in follow-ups once we have API evolution in place

Reply:
I figured this was best addressed by a FAQ:

> Why is there a new APIResourceSchema resource type that appears to be very similar to CustomResourceDefinition?

and honestly I got a bit stuck coming up with a satisfying answer ... like one that differentiates APIResourceSchema as capturing different versions of a CRD versus, uh, CRD versions.

I think it's best to drop a FIXME in here for this FAQ and do a follow-up docs update where we explain the rationale for it in some depth. Ideally the answer to this FAQ would be a link to design doc